src/sql/workbench/browser/modelComponents/radioCardGroup.component.ts

```diff
@@ -0,0 +1,179 @@
+			if (this.focusedCard) {
+				this.selectCard(this.findNextCard(this.focusedCard));
+			}
+			DOM.EventHelper.stop(e, true);
+		}
+	}
+
+	private findPreviousCard(currentCard: azdata.RadioCard): azdata.RadioCard {
+		const currentIndex = this.cards.indexOf(currentCard);
+		const previousCardIndex = currentIndex === 0 ? this.cards.length - 1 : currentIndex - 1;
+		return this.cards[previousCardIndex];
+	}
+
+	private findNextCard(currentCard: azdata.RadioCard): azdata.RadioCard {
+		const currentIndex = this.cards.indexOf(currentCard);
+		const nextCardIndex = currentIndex === this.cards.length - 1 ? 0 : currentIndex + 1;
+		return this.cards[nextCardIndex];
+	}
+
+	public get cards(): azdata.RadioCard[] {
+		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, azdata.RadioCard[]>((props) => props.cards, []);
+	}
+
+	public get cardWidth(): string | undefined {
+		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, string | undefined>((props) => props.cardWidth, undefined);
+	}
+
+	public get cardHeight(): string | undefined {
+		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, string | undefined>((props) => props.cardHeight, undefined);
+	}
+
+	public get iconWidth(): string | undefined {
+		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, string | undefined>((props) => props.iconWidth, undefined);
+	}
+
+	public get iconHeight(): string | undefined {
+		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, string | undefined>((props) => props.iconHeight, undefined);
+	}
+
+	public get selectedCardId(): string | undefined {
+		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, string | undefined>((props) => props.selectedCardId, undefined);
+	}
+
+	public set selectedCardId(newValue: string | undefined) {
+		this.setPropertyFromUI<azdata.RadioCardGroupComponentProperties, string | undefined>((props, value) => props.selectedCardId = value, newValue);
+	}
+
+	public getIconClass(card: azdata.RadioCard): string {
+		if (!this.iconClasses[card.id]) {
+			this.iconClasses[card.id] = `cardIcon icon ${getIconClass(card.icon)}`;
+		}
+		return this.iconClasses[card.id];
+	}
+
+	public setProperties(properties: { [key: string]: any }) {
+		super.setProperties(properties);
+		if (this.selectedCardId) {
+			const filteredCards = this.cards.filter(c => { return c.id === this.selectedCardId; });
+			if (filteredCards.length === 1) {
+				this.selectCard(filteredCards[0]);
+			}
+		}
+	}
+
+	public selectCard(card: azdata.RadioCard): void {
+		if (!this.enabled || this.selectedCard === card || this.cards.indexOf(card) === -1) {
+			return;
+		}
+		this.selectedCard = card;
+		this._changeRef.detectChanges();
+		const cardElement = this.getCardElement(this.selectedCard);
+		cardElement.nativeElement.focus();
+		this.selectedCardId = card.id;
```

Comment:
if we already have a this.selectedCard, do we need to keep track of its id in a different var as well? Seems rife for a sneaky bug

Reply:
hmm, the selectedCardId is for the extension to use and the this.selectedCard is the internal state of the card group component. I won't be able to simply do obj assign since for objects it is reference based comparison.